test/compiler/inline.jl

```diff
@@ -1301,8 +1301,28 @@ let src = code_typed1(Tuple{Any}) do x
             DoAllocNoEscapeSparam(x)
         end
     end
+    @test count(x->isexpr(x, :static_parameter), src.code) == 0 # A bad inline might leave left-over :static_parameter
+    nnothrow_invokes = count(isinvoke(:nothrow_side_effect), src.code)
+    @test count(iscall(f->!isa(singleton_type(argextype(f, src)), Core.Builtin)), src.code) ==
+          count(iscall((src, nothrow_side_effect)), src.code) == 2 - nnothrow_invokes
     # FIXME
```

Comment:
can you add a comment to this about what is bad here?

Reply:
It's just a suboptimality, but we had an existing test for it that turned out to be broken. There's 2-3 more issues that prevent this from being eliminated, but I intend to get through them this week.